server/routerlicious/packages/lambdas-driver/src/document-router/documentContext.ts

```diff
@@ -79,15 +81,44 @@ export class DocumentContext extends EventEmitter implements IContext {
 		// Assert offset is between the current tail and head
 		const offset = message.offset;
 
-		assert(
-			offset > this.tail.offset && offset <= this.head.offset,
-			`${offset} > ${this.tail.offset} && ${offset} <= ${this.head.offset} ` +
-				`(${message.topic}, ${message.partition}, ${this.routingKey.tenantId}/${this.routingKey.documentId})`,
-		);
-
-		// Update the tail and broadcast the checkpoint
-		this.tailInternal = message;
-		this.emit("checkpoint", restartOnCheckpointFailure);
+		try {
+			assert(
+				offset > this.tail.offset && offset <= this.head.offset,
+				`${offset} > ${this.tail.offset} && ${offset} <= ${this.head.offset} ` +
+					`(${message.topic}, ${message.partition}, ${this.routingKey.tenantId}/${this.routingKey.documentId})`,
+			);
+
+			// Update the tail and broadcast the checkpoint
+			this.tailInternal = message;
+			this.emit("checkpoint", restartOnCheckpointFailure);
+		} catch (error) {
```

Comment:
These asserts should only be fired when something catastrophic occurs, such as kafka consuming messages in the wrong order.
Do you know why the assert is firing exceptions? I'm curious why you're seeing these errors.

Reply:
After discussing with @pradeepvairamani, we've decided to find the root-cause of the issue rather than catching the assertion error, since we agree that we should not allow this case to happen. I will close this pull request for now